client/src/components/molecules/RequestsTable.tsx

```diff
@@ -0,0 +1,82 @@
+import { gql, useMutation } from '@apollo/client';
+import React, { FunctionComponent,  useState } from 'react';
+import  Form  from 'react-bootstrap/Form';
+import moment from 'moment';
+import Request from '../../data/types/request';
+import  Table  from 'react-bootstrap/Table';
+
+interface Props {
+    requests: Request[];
+}
+
+const RequestsTable: FunctionComponent<Props> = (props: Props) => {
+    const headingList = ['Fulfilled', 'Client Name', 'Quantity', 'Date Requested', '', ''];
+    const updateRequest = gql`
+    mutation updateRequest($request: RequestInput){
+        updateRequest(request: $request){
+          id
+          success
+          message
+        }
+      }
+    `;
+
+    const [requests, setRequests] = useState(props.requests);
+    const [mutateRequest] = useMutation(updateRequest);
+    console.log(requests);
+
+    const onFulfilledRequest = (index:number) => {
+        const requestsCopy = requests.slice();
+        const req = {...requestsCopy[index]};
+        req.fulfilled = !req.fulfilled;
+        requestsCopy[index]=req
+        setRequests(requestsCopy);
+        const id = req._id;
+        const requestId = req.requestId;
+        const fulfilled = req.fulfilled;
+        mutateRequest({variables:{request: {id, requestId, fulfilled}}});
+    }
+
+    
+    return (
+        <div className="request-list">
+            <Table responsive className="request-table">
+                <thead>
+                <tr className="request-table header-style">
+                    {headingList.map((heading, index) => (
```

Comment:
Use of variables is unnecessary here. Maybe just put it inline so its easier to modify later if needed or add specific actions to specific columns (e.g. if we have sorting later).

Reply:
I'm not sure what you mean. I just thought having the headers in a list variable would be good because then for sorting you can modify the variable

Reply:
For example, header has fields "Quantity", "Client Name". For sorting, perhaps when you click on some button on the header for "Quantity", it sorts by number ascending, but when you click on some button on the header for "Client Name" it sorts by client's last name ascending. Clicking on "Client Name" sorting button again now sorts it in descending.

We can't do this unless we can modify each header individually.

Reply:
You can have an onClick on the header and pass in the index to the handleClick function. Based on what the index is we can handle the sort accordingly